tasks/poker.ts

```diff
@@ -0,0 +1,174 @@
+        const query = gql`
+            {
+                boostedSavingsVaults {
+                    id
+                    boostCoeff
+                    priceCoeff
+                    stakingToken {
+                        symbol
+                    }
+                    accounts(where: { rawBalance_gt: "0" }) {
+                        account {
+                            id
+                        }
+                        rawBalance
+                        boostedBalance
+                        rewardPerTokenPaid
+                        rewards
+                        lastClaim
+                        lastAction
+                    }
+                }
+                _meta {
+                    block {
+                        number
+                    }
+                }
+            }
+        `
+        const gqlData = await gqlClient.request(query)
+
+        // eslint-disable-next-line no-underscore-dangle
+        const blockNumber = gqlData._meta.block.number
+        console.log(`Results for block number ${blockNumber}`)
+
+        // Maps GQL to a list if accounts (addresses) in each vault
+        const vaultAccounts = gqlData.boostedSavingsVaults.map((vault) => vault.accounts.map((account) => account.account.id))
+        const accountsWithDuplicates = vaultAccounts.flat()
+        const accountsUnique = [...new Set<string>(accountsWithDuplicates)]
+        const vMtaBalancesMap = await getAccountBalanceMap(accountsUnique, MTA.saving, signer)
+        const pokeVaultAccounts: {
+            boostVault: string
+            accounts: string[]
+        }[] = []
+        // For each Boosted Vault
+        for (const vault of gqlData.boostedSavingsVaults) {
+            const boostVault = BoostedSavingsVault__factory.connect(vault.id, signer)
+            const priceCoeff = await boostVault.priceCoeff()
+            const boostCoeff = await boostVault.boostCoeff()
+
+            const overBoosted: any[] = []
+            console.log(
+                `\nVault with id ${vault.id} for token ${vault.stakingToken.symbol}, ${vault.accounts.length} accounts, price coeff ${priceCoeff}, boost coeff ${boostCoeff}`,
+            )
+            console.log("Account, Raw Balance, Boosted Balance, vMTA balance, vMTA diff, Boost Actual, Boost Expected, Boost Diff")
+            vault.accounts.forEach((account, i) => {
+                const boostActual = BN.from(account.boostedBalance).mul(1000).div(account.rawBalance).toNumber()
+                const boostExpected = calcBoost(BN.from(account.rawBalance), vMtaBalancesMap[account.account.id], priceCoeff, boostCoeff)
+                    .div(BN.from(10).pow(15))
+                    .toNumber()
+                const boostDiff = boostActual - boostExpected
+                // Calculate how much extra MTA is being distributed = vMTA balance * (actual boost - expected boost)
+                const vMtaExtra = vMtaBalancesMap[account.account.id].mul(boostDiff).div(1000)
```

Comment:
Actually I think this needs to be fixed. Imagine a users vMTA balance is now 0 - they could have a 3x boost on $1m and it would be filtered out here.

It makes much more sense to detect the value of a users boosted holdings - you can just do `account.boostedBalance * priceCoeff / 1e18` to find the USD value of their boost. I would say, if `boostDiff > 200 && USD balance > 50k` then poke.